### PR TITLE
Consider checksums in recipe conflicts

### PR DESCRIPTION
**Pull request by ilkka**
When a new recipe version modifies a file which has not been changed by the user since rendering, it should be overwritten without asking when doing an upgrade. This requires adding checking file checksums (which we already save when rendering).

### CODE REVIEW

File: cmd/upgrade.go [hunk 1]
```diff
@@ -130,35 +129,33 @@ func upgradeFunc(cmd *cobra.Command, args []string) {
...
+
+				// TODO: We could do better in terms of merge conflict management. Like show the diff or something
+				var override bool
+				prompt := &survey.Confirm{
+					Message: path,
+					Default: true,
+				}
+				err = survey.AskOne(prompt, &override)
+				if err != nil {
+					cmd.Println(err)
```
Comment (majori) [inline cmd/upgrade.go:150]: Should we use `cmd.Printerr()` here?

File: cmd/upgrade.go [hunk 2]
```diff
@@ -130,35 +129,33 @@ func upgradeFunc(cmd *cobra.Command, args []string) {
...
-			var override bool
-			prompt := &survey.Confirm{
-				Message: path,
-				Default: true,
-			}
-			err = survey.AskOne(prompt, &override)
-			if err != nil {
+			// Check if file was modified after rendering
+			if modified, err := recipeutil.IsFileModified(target, path, prevFile); err != nil {
 				cmd.Println(err)
```
Comment (majori) [inline cmd/upgrade.go:134]: Should we use cmd.Printerr() here?

File: cmd/cmds_test.go [hunk 3]
```diff
@@ -224,6 +224,24 @@ func noConflictsWereReported(ctx context.Context) (context.Context, error) {
 	return ctx, nil
 }
 
+func conflictsAreReported(ctx context.Context) (context.Context, error) {
+	recipeStdout := ctx.Value(recipeStdoutCtxKey{}).(string)
+	recipeStderr := ctx.Value(recipeStderrCtxKey{}).(string)
+	if matched, _ := regexp.Match("modified", []byte(recipeStdout)); matched {
```
Comment (majori) [inline cmd/cmds_test.go:230]: `regexp.MatchString` could be used to avoid extra casting

File: cmd/cmds_test.go [hunk 4]
```diff
@@ -224,6 +224,24 @@ func noConflictsWereReported(ctx context.Context) (context.Context, error) {
 	return ctx, nil
 }
 
+func conflictsAreReported(ctx context.Context) (context.Context, error) {
+	recipeStdout := ctx.Value(recipeStdoutCtxKey{}).(string)
+	recipeStderr := ctx.Value(recipeStderrCtxKey{}).(string)
```
Comment (majori) [inline cmd/cmds_test.go:229]: Should we have a step which checks that stderr is empty?

**Review by majori — APPROVED** [review]

File: cmd/cmds_test.go (hunk 4 above)
Comment (ilkka) [reply cmd/cmds_test.go:229]: Added in a8aebdce085ab68086b6e30182879899a888cf3f

**Pull request merged**